languages/typescript/src/main/kotlin/io/vrap/codegen/languages/typescript/joi/JoiValidatorModuleRenderer.kt

```diff
@@ -207,4 +207,25 @@ class JoiValidatorModuleRenderer constructor(override val vrapTypeProvider: Vrap
             }
         }
     }
+
+    private fun EList<Property>.renderPatternProperties(): String {
+        return this.filter { it.isPatternProperty() }
+            .joinToString(separator = "\n") {
+                renderTypes(it)
+            }
+        }
+
+    private fun renderTypes(it: Property) = when (it.type) {
+        is UnionType -> renderUnion(it)
+        else -> renderDefault(it)
+    }
+
+    private fun renderUnion(it: Property) = ".pattern(${it.asRegExp()}, ${
+        (it.type as UnionType).oneOf.map { ite ->
+            if (ite is NilType) "null" else it.type.toVrapType().renderTypeRef()
```

Comment:
wouldn't it make more sense to say here `.allow(null)` if one of the union types can be null?

Reply:
Hi Jens, 
Maybe so, where would I put the .allow(null)? I looked all over the code and didn't see any references.

Reply:
I'm no expert for joi DSL :) But I would at least try to build the schema validator using the DSL functions

Reply:
Besides I think checking for NilType in the UnionType will not really help as NilType is a special type. I think what you more likely want to achieve is checking if the property itself is optional.

Reply:
But the thing is, the setting that I want to render is `type: Attribute | nil`, in this case a map item comes as NilType.
And the idea is to render to `Joi.object().pattern(/.*/, [attributeType(), null])`, in which case it would be a direct rendering of nil.